salt/client/mixins.py

```diff
@@ -397,11 +397,15 @@ def _low(self, fun, low, print_event=True, full_return=False):
             if isinstance(ex, salt.exceptions.NotImplemented):
                 data['return'] = six.text_type(ex)
             else:
-                data['return'] = 'Exception occurred in {0} {1}: {2}'.format(
-                    self.client,
-                    fun,
-                    traceback.format_exc(),
-                    )
+                exc_msg = six.text_type(ex)
+                if 'Exiting gracefully' in exc_msg:
```

Comment:
I think there is a better way of doing this, something like redefine exception class in `scripts/_handle_signals` like so:

```python
if signum == signal.SIGINT:
    exception_class = SaltSystemExit  # Or something else?
    exit_msg = '\nExiting gracefully on Ctrl-C'
    ...
else:
    exception_class = SystemExit

_handle_interrupt(exception_class(exit_msg)......
```

And then capture it properly in a separate catch?

Reply:
Interesting thought. @terminalmage ?